contracts/utils/EnumerableSet.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity ^0.5.10;
+
+
+/**
+ * @title EnumerableSet
+ * @dev Data structure - https://en.wikipedia.org/wiki/Set_(abstract_data_type)
+ *
+ * An EnumerableSet.AddressSet is a data structure containing a number of unique addresses.
+ *
+ *  - It is possible to add and remove addresses in O(1).
+ *  - It is also possible to query if the AddressSet contains an address in O(1).
+ *  - It is possible to retrieve an array with all the addresses in the AddressSet using enumerate.
+ *    This operation is O(N) where N is the number of addresses in the AddressSet. The order in
+ *    which the addresses are retrieved is not guaranteed.
+ *
+ * Initialization of a set must include an empty array:
+ * `EnumerableSet.AddressSet set = EnumerableSet.AddressSet({values: new address[](0)});`
+ *
+ * @author Alberto Cuesta Cañada
+ */
+library EnumerableSet {
+
+    event ValueAdded(address value);
+    event ValueRemoved(address value);
+
+    struct AddressSet {
+        // Position of the value in the `values` array, plus 1 because index 0
+        // means a value is not in the set.
+        mapping (address => uint256) index;
+        address[] values;
+    }
+
+    /**
+     * @dev Add a value. O(1).
+     */
+    function add(AddressSet storage set, address value)
+        internal
+    {
+        if (!contains(set, value)){
```

Comment:
I'm not sure if we should revert in this scenario. In `Roles` for example we require that the account doesn't already have the role when it is added. 

At the same time, in both cases the result will be the same: the address will be added to the set. Thoughts?

Reply:
Alternatively, we could return a `bool`, indicating whether or not the operation actually did anything, like [`Rust does`](https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.insert).

Reply:
I always prefer to give a consistent behaviour to the users. All three options are fine, but if you are reverting on useless operations in the rest of the package, we should do the same here.